src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1020,7 +1020,7 @@ private function registerWorkflowConfiguration(array $config, ContainerBuilder $
                     throw new LogicException('Cannot guard workflows as the ExpressionLanguage component is not installed. Try running "composer require symfony/expression-language".');
                 }
 
-                if (!class_exists(Security::class)) {
+                if (!class_exists(AuthenticationEvents::class)) {
```

Comment:
```suggestion
                if (!class_exists(AuthenticationException::class)) {
```

This is just a check whether the Security component is installed. I think it's better to choose a class that (a) is actually used in this bundle and (b) exists in all supported versions.

Reply:
Given it's ok for (b): https://github.com/symfony/symfony/blob/2424b43320b14df04830e3dedaa2fa2daeb4645a/src/Symfony/Bundle/FrameworkBundle/composer.json#L92

Should we really care about (a) given this is just a package existence check?

Reply:
Well, the main reason for the switch to me is that the existence check should use a non-deprecated class, so that (b) stays true when reaching 7.0 where the class would be removed. A deprecated class is not future-proof for (b).

Reply:
All good with `AuthenticationEvents` then given it's not deprecated, right?

Reply:
yes, indeed